bool[4] obstacle